packages/inline-edit/src/InlineEditUncontrolled.tsx

```diff
@@ -0,0 +1,208 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import React, {
+  FC, ReactNode, useRef, useCallback, createRef, useState, useEffect,
+} from 'react';
+
+import {
+  ReadViewWrapper,
+  EditButton,
+  ReadViewContentWrapper,
+  ButtonsWrapper,
+  ButtonWrapper,
+  Button,
+  ContentWrapper,
+} from './styled/InlineEdit';
+import {
+  ConfirmIcon,
+  CancelIcon,
+} from './Icons';
+import { IInlineEditUncontrolledProps } from './interfaces';
+
+
+export const getInlineEditUncontrolled: <TFieldValue = string>() => FC<
+IInlineEditUncontrolledProps<TFieldValue>> = function f<TFieldValue>() {
+  return ({
+    appearance,
+    baseAppearance,
+    defaultValue,
+    readView,
+    editView,
+    readViewFitContainerWidth,
+    isEditing,
+    onEditRequested,
+    onCancel,
+    onConfirm,
+  }: IInlineEditUncontrolledProps<TFieldValue>) => {
+    const editButtonRef = createRef<HTMLButtonElement>();
+    const confirmButtonRef = createRef<HTMLButtonElement>();
+    const cancelButtonRef = createRef<HTMLButtonElement>();
+
+    const startX = useRef(0);
+    const startY = useRef(0);
+    const prevIsEditing = useRef(isEditing);
+    const removeMouseDown: React.MutableRefObject<void | null> = useRef(null);
+
+    const [value, setValue] = useState(defaultValue);
+    const [preventFocusOnEditButton, setPreventFocusOnEditButton] = useState(false);
+
+    useEffect(() => {
+      if (prevIsEditing.current && !isEditing) {
+        if (preventFocusOnEditButton) {
+          setPreventFocusOnEditButton(false);
+        } else if (editButtonRef && editButtonRef.current) {
+          editButtonRef.current.focus();
+        }
+      }
+    }, [editButtonRef, isEditing, preventFocusOnEditButton]);
+
+    useEffect(() => {
+      function handleMouseDown(this: Document, e: MouseEvent): void {
+        if (confirmButtonRef.current?.contains(e.target as Node)) {
+          onConfirm(value);
+        } else if (cancelButtonRef.current?.contains(e.target as Node)) {
+          onCancel();
+        }
+      }
+
+      if (typeof removeMouseDown.current === 'function') {
+        document.removeEventListener('mousedown', handleMouseDown);
+        removeMouseDown.current = null;
+      }
+
+      removeMouseDown.current = document.addEventListener(
+        'mousedown',
+        handleMouseDown,
+        { capture: true },
+      );
+    }, [cancelButtonRef, confirmButtonRef, onCancel, onConfirm, value]);
+
+    const mouseHasMoved = useCallback((event: { clientX: number; clientY: number }): boolean => (
+      Math.abs(startX.current - event.clientX) >= 5 || Math.abs(startY.current - event.clientY) >= 5
+    ), []);
+
+    const handleEditValueChange = useCallback((e) => {
```

Comment:
`handleEditValueChange` is passed in `editView`.
`editView` should pass in `handleEditValueChange ` final value and `editView ` must know how to get correct value.
I mean that it is incorrect to pass in this function event.